src/Razor/src/rzls/LogLevelProvider.cs

```diff
@@ -0,0 +1,18 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System.Threading;
+using Microsoft.CodeAnalysis.Razor.Logging;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer;
+
+internal class LogLevelProvider(LogLevel logLevel)
+{
+    private int _logLevel = (int)logLevel;
+
+    internal LogLevel GetLogLevel()
+        => (LogLevel)_logLevel;
+
+    internal void SetLogLevel(LogLevel logLevel)
+        => Interlocked.Exchange(ref _logLevel, (int)logLevel);
```

Comment:
Why is this field assignment using Interlocked.Exchamge()? Is it expected that there will be a lot of contention? Does it matter if GetLogLevel() sometimes returns the wrong answer?

Reply:
Defensive programming. This can just be a property